### DISCUSSION TITLE
When will the stable (not beta) version of kotlin sdk be released?

### DISCUSSION THREAD

**Discussion by sugoikondo** [aws/aws-sdk-kotlin#929] [General]
Hi, Thanks for providing excellent SDK implementation for Kotlin!
(Forgive me that my English sentence is awful, so sorry if it is too difficult to understand)

I'm awaiting a stable release of aws-kotlin-sdk.
If I'm correct, the last not-beta version release was v0.16.0. Since it, only beta version of SDKs are released.

So, do you have plans to release a stable or not-beta version of sdk in the near future?
Or, is beta release versions are already on a level that can be used on production?

I'm so sorry to disturb you, but I'd appreciate it if you let me know the current status of development!

**Comment by ianbotsf** [comment]
We're working hard towards a stable, 1.0 release of the AWS SDK for Kotlin. To that end, we're focusing on stabilizing our APIs, adding in features for the most common use cases, and optimizing performance so that it's as reliable as possible. At this time we cannot commit to a firm date for a 1.0 release but it's our top priority!

Every release up to now has been an alpha or beta release, even **v0.16.0** (which was mistakenly named without the **-beta** suffix). Our releases will only be considered non-beta once they go to 1.0. As [noted in our **README.md**](https://github.com/awslabs/aws-sdk-kotlin/blob/main/README.md#aws-sdk-for-kotlin), we do not recommend using this SDK for production workloads yet. Given that some features may be missing and some code paths may be unoptimized, usage of the SDK in production is undertaken at your own risk. Additionally, we may continue to make breaking API changes which may complicate upgrading to later versions.

That said, we believe that each release of the SDK is an improvement on what came before. We encourage you to try it out as a beta dependency in non-critical environments provide us feedback on what works well and what still needs more effort. We rely on that feedback to help us determine if/when we're ready for a 1.0 release.

**Reply by sugoikondo** [reply]
Thank you very much for your quick and gracious response!!
And so glad to hear that team is working on the release of a stable version!

> Every release up to now has been an alpha or beta release, even v0.16.0  (which was mistakenly named without the -beta suffix)

Ah, I see!
I'm sorry for my misunderstanding ... I mistakenly thought it was a somewhat stable version.

And, Once again, thank you so much for your continued work on the stable release!
I've started using the kotlin SDK on some small applications development, so I would appreciate if you let me give you some feedback if I found it!

**Reply by ianbotsf** [reply]
> I've started using the kotlin SDK on some small applications development, so I would appreciate if you let me give you some feedback if I found it!

Please do! If you find bugs or missing features/docs, please [file an issue with us](https://github.com/awslabs/aws-sdk-kotlin/issues/new/choose). If you find anything confusing or you want to have a conversation about something, please [start a new discussion](https://github.com/awslabs/aws-sdk-kotlin/discussions/new/choose).
